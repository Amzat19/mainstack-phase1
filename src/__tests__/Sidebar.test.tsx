import { render, screen } from '@testing-library/react';
import Sidebar from '../components/SideBar';
import '@testing-library/jest-dom';

jest.mock('../assets/mainstack-logo.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});

jest.mock('../assets/dashboard.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});

jest.mock('../assets/edit.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});

jest.mock('../assets/hourglass_empty.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});

jest.mock('../assets/add_a_photo.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});
jest.mock('../assets/delete.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});
jest.mock('../assets/subscriptions.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});
jest.mock('../assets/file_present.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});

jest.mock('../assets/alarm.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});

jest.mock('../assets/more_horiz.svg', () => {
  return {
    ReactComponent: () => <svg>Mocked Info SVG</svg>,
  };
});



describe('SideBar component', () => {
  test('renders without errors', () => {
    render(<Sidebar />);
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Sidebar />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('displays the dashboard icon and label', () => {
     render(<Sidebar />);
    const dashboardHeaderText = screen.getByText('Dashboard');
    expect(dashboardHeaderText).toBeInTheDocument();
    expect(dashboardHeaderText.tagName.toLowerCase()).toBe('p');
    // Add more assertions as needed
  });

  test('displays the item icons and labels', () => {
    const { getByText } = render(<Sidebar />);
    const item1Icon = getByText('Item 1');
    const item2Icon = getByText('Item 2');
    const item3Icon = getByText('Item 3');

    expect(item1Icon).toBeInTheDocument();
    expect(item2Icon).toBeInTheDocument();
    expect(item3Icon).toBeInTheDocument();
    // Add more assertions as needed
  });

});