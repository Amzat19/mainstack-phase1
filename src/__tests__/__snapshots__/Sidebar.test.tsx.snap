// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component matches snapshot 1`] = `
<DocumentFragment>
  <aside
    class="sidebar"
  >
    <div
      class="logo"
    >
      <svg>
        Mocked Info SVG
      </svg>
    </div>
    <div
      class="sideLinks"
    >
      <div>
        <span
          class="activePage"
        >
          <svg>
            Mocked Info SVG
          </svg>
          <p>
            Dashboard
          </p>
        </span>
        <ul>
          <li>
            <svg>
              Mocked Info SVG
            </svg>
            <p>
              Item 1
            </p>
          </li>
          <li>
            <svg>
              Mocked Info SVG
            </svg>
            <p>
              Item 2
            </p>
          </li>
          <li>
            <svg>
              Mocked Info SVG
            </svg>
            <p>
              Item 3
            </p>
          </li>
        </ul>
      </div>
      <div>
        <span>
          Others1
        </span>
        <ul>
          <li>
            <svg>
              Mocked Info SVG
            </svg>
            <p>
              Item 4
            </p>
          </li>
          <li>
            <svg>
              Mocked Info SVG
            </svg>
            <p>
              Item 5
            </p>
          </li>
        </ul>
      </div>
      <div>
        <span>
          Other2
        </span>
        <ul>
          <li>
            <svg>
              Mocked Info SVG
            </svg>
            <p>
              Item 7
            </p>
          </li>
          <li>
            <svg>
              Mocked Info SVG
            </svg>
            <p>
              Item 8
            </p>
          </li>
          <li>
            <svg>
              Mocked Info SVG
            </svg>
            <p>
              Item 9
            </p>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="profile"
    >
      <img
        alt="a dark skin woman"
        src="[object Object]"
      />
      <p>
        Blessing Daniels
      </p>
      <svg>
        Mocked Info SVG
      </svg>
    </div>
  </aside>
</DocumentFragment>
`;
